client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -129,6 +129,7 @@ public class TrinoConnection
     private final Map<String, String> sessionProperties = new ConcurrentHashMap<>();
     private final Map<String, String> preparedStatements = new ConcurrentHashMap<>();
     private final Map<String, ClientSelectedRole> roles = new ConcurrentHashMap<>();
+    private Map<String, Class<?>> typeMap = new ConcurrentHashMap<>();
```

Comment:
If the variable value is only set via setTypeMap, then we can initialize it to `null`